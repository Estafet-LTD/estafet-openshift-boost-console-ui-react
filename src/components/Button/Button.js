import styled, { css } from 'styled-components';
import { ifProp, switchProp, withProp } from 'styled-tools';

import { baseColor } from '../../styles/variables/colors';

export default styled.button`
  outline: none;
  cursor: pointer;
  border: 0;
  height: 1.8rem;
  line-height: 1.8rem;
  border-radius: 3px;
  font-size: 0.9rem;
  padding: 0 0.5rem;
  transition: 150ms all ease-in;
  color: ${switchProp('type', {
    primary: 'white',
    secondary: baseColor,
  })};
  background: ${switchProp('type', {
    primary: baseColor,
    secondary: 'transparent',
  })};
  border: ${withProp(
    switchProp('type', {
      primary: 'transparent',
      secondary: baseColor,
    }),
    (color) => `1px solid ${color}`
  )};

  &:hover {
    color: white;
    background: ${switchProp('type', {
      primary: '#138496',
      secondary: baseColor,
    })};
  }

  ${ifProp(
    'isDisabled',
    css`
      pointer-events: none;
      /*
      color: #bbbbbb;
      background: #dedede;
      border-color: #bbbbbb;
      */
    `
  )};
`;
