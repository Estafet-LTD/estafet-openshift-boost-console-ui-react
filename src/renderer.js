// TODO: Clean up and split up all the different logic handling styled-components, variables set on window object, etc
import { ChunkExtractor, ChunkExtractorManager } from '@loadable/server';
import path from 'path';
import React from 'react';
import { renderToString } from 'react-dom/server';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router-dom';
import { ServerStyleSheet, StyleSheetManager } from 'styled-components';

import App from './App';
import configureStore from './store';

const store = configureStore();
const sheet = new ServerStyleSheet();

const statsFile = path.resolve('build/loadable-stats.json');
const extractor = new ChunkExtractor({
  statsFile,
  entrypoints: ['client'],
});

export default (req, res) => {
  const context = {};
  const html = renderToString(
    <ChunkExtractorManager extractor={extractor}>
      <StaticRouter context={context} location={req.url}>
        <Provider store={store}>
          <StyleSheetManager sheet={sheet.instance}>
            <App />
          </StyleSheetManager>
        </Provider>
      </StaticRouter>
    </ChunkExtractorManager>
  );

  const styleTags = sheet.getStyleTags();
  const scriptTags = extractor.getScriptTags();

  if (context.url) {
    res.redirect(context.url);
  } else {
    res.status(200).send(
      `<!doctype html>
      <html lang="">
        <head>
          <meta charset="utf-8" />
          <title>${process.env.PRODUCT_DESCRIPTION}</title>
          <meta name="description" content="" />
          <meta name="theme-color" content="#2BA2B8" />
          <meta name="viewport" content="width=device-width, initial-scale=1">
          <link href="https://fonts.googleapis.com/css?family=Open+Sans&display=swap" rel="stylesheet">
          ${styleTags}
        </head>
        <body>
          <div id="root">${html}</div>
          ${scriptTags}
          <script>
            window.__PRELOADED_STATE__ = ${JSON.stringify(store.getState())};
            window.boost = ${JSON.stringify({
              BUILD_API_SERVICE_URI: process.env.BUILD_API_SERVICE_URI,
              TEST_API_SERVICE_URI: process.env.TEST_API_SERVICE_URI,
              PROD_API_SERVICE_URI: process.env.PROD_API_SERVICE_URI,
              PRODUCT_DESCRIPTION: process.env.PRODUCT_DESCRIPTION,
            })};
          </script>
        </body>
      </html>`
    );
  }
};
