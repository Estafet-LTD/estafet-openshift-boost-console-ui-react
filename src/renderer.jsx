import { ChunkExtractor, ChunkExtractorManager } from '@loadable/server';
import path from 'path';
import React from 'react';
import { renderToString } from 'react-dom/server';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router-dom';
import { ServerStyleSheet, StyleSheetManager } from 'styled-components';

import App from './App';
import configureStore from './store/configureStore';

const store = configureStore();
const sheet = new ServerStyleSheet();

// Mocking the loadable stats file for the tests
const isTesting = process.env.NODE_ENV === 'test';
const statsFileRoot = isTesting ? 'src/__mocks__' : 'build';
const statsFile = path.resolve(`${statsFileRoot}/loadable-stats.json`);

const extractor = new ChunkExtractor({
  statsFile,
  entrypoints: ['client'],
});

export default (req, res) => {
  const context = {};
  const html = renderToString(
    <ChunkExtractorManager extractor={extractor}>
      <StaticRouter context={context} location={req.url}>
        <Provider store={store}>
          <StyleSheetManager sheet={sheet.instance}>
            <App />
          </StyleSheetManager>
        </Provider>
      </StaticRouter>
    </ChunkExtractorManager>
  );

  const styledComponentsTags = sheet.getStyleTags();
  const styleTags = extractor.getStyleTags();
  const scriptTags = extractor.getScriptTags();

  if (context.url) {
    res.redirect(context.url);
  } else {
    res.status(200).send(
      `<!doctype html>
      <html lang="en">
        <head>
          <meta charset="utf-8" />
          <title>${process.env.PRODUCT_DESCRIPTION}</title>
          <meta name="description" content="" />
          <meta name="theme-color" content="#2BA2B8" />
          <meta name="viewport" content="width=device-width, initial-scale=1">
          <link href="https://fonts.googleapis.com/css?family=Open+Sans:400,700&display=swap" rel="stylesheet">
          ${styleTags}
          ${styledComponentsTags}
        </head>
        <body>
          <div id="root">${html}</div>
          ${scriptTags}
          <script>
            window.__PRELOADED_STATE__ = ${JSON.stringify(store.getState())};
            window.__GLOBALS__= {
              PRODUCT: ${JSON.stringify(process.env.PRODUCT)},
              PRODUCT_DESCRIPTION: ${JSON.stringify(process.env.PRODUCT_DESCRIPTION)},
              TASK_MANAGEMENT_TITLE: ${JSON.stringify(process.env.TASK_MANAGEMENT_TITLE)},
              GATEWAY_API_SERVICE_URI: ${JSON.stringify(process.env.GATEWAY_API_SERVICE_URI)},
            };
          </script>
        </body>
      </html>`
    );
  }
};
