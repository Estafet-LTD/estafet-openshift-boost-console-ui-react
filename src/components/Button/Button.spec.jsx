import React from 'react';
import renderer from 'react-test-renderer';

import Button from './Button';

describe('Components: Button', () => {
  it('it should match the snapshot [primary]', () => {
    const tree = renderer.create(<Button variant='primary'>Primary button</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('it should match the snapshot [secondary]', () => {
    const tree = renderer.create(<Button variant='secondary'>Secondary button</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('it should match the snapshot [preventEvents]', () => {
    const tree = renderer
      .create(
        <Button variant='primary' preventEvents>
          Primary button with disabled events
        </Button>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('it should match the snapshot [isDisabled]', () => {
    const tree = renderer.create(<Button isDisabled>Disabled button</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
