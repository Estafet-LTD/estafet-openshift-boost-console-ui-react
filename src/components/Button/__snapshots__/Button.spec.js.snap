// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: Button it should match the snapshot [isDisabled] 1`] = `
.c0 {
  outline: none;
  cursor: pointer;
  border: 0;
  height: 1.8rem;
  line-height: 1.8rem;
  border-radius: 3px;
  font-size: 0.9rem;
  padding: 0 0.5rem;
  -webkit-transition: 150ms all ease-in;
  transition: 150ms all ease-in;
  border: 1px solid undefined;
  pointer-events: none;
}

.c0:hover {
  color: white;
}

<button
  className="c0"
>
  Disabled button
</button>
`;

exports[`Components: Button it should match the snapshot [primary] 1`] = `
.c0 {
  outline: none;
  cursor: pointer;
  border: 0;
  height: 1.8rem;
  line-height: 1.8rem;
  border-radius: 3px;
  font-size: 0.9rem;
  padding: 0 0.5rem;
  -webkit-transition: 150ms all ease-in;
  transition: 150ms all ease-in;
  color: white;
  background: #2BA2B8;
  border: 1px solid transparent;
}

.c0:hover {
  color: white;
  background: #138496;
}

<button
  className="c0"
>
  Primary button
</button>
`;

exports[`Components: Button it should match the snapshot [secondary] 1`] = `
.c0 {
  outline: none;
  cursor: pointer;
  border: 0;
  height: 1.8rem;
  line-height: 1.8rem;
  border-radius: 3px;
  font-size: 0.9rem;
  padding: 0 0.5rem;
  -webkit-transition: 150ms all ease-in;
  transition: 150ms all ease-in;
  color: #2BA2B8;
  background: transparent;
  border: 1px solid #2BA2B8;
}

.c0:hover {
  color: white;
  background: #2BA2B8;
}

<button
  className="c0"
>
  Secondary button
</button>
`;
